motionplan/motionPlanner.go

```diff
@@ -27,31 +26,32 @@ import (
 type motionPlanner interface {
 	// Plan will take a context, a goal position, and an input start state and return a series of state waypoints which
 	// should be visited in order to arrive at the goal while satisfying all constraints
-	plan(context.Context, spatialmath.Pose, []frame.Input) ([]node, error)
+	plan(context.Context, PathStep, map[string][]referenceframe.Input) ([]node, error)
 
 	// Everything below this point should be covered by anything that wraps the generic `planner`
 	smoothPath(context.Context, []node) []node
-	checkPath([]frame.Input, []frame.Input) bool
-	checkInputs([]frame.Input) bool
-	getSolutions(context.Context, []frame.Input) ([]node, error)
+	checkPath(map[string][]referenceframe.Input, map[string][]referenceframe.Input) bool
+	checkInputs(map[string][]referenceframe.Input) bool
+	getSolutions(context.Context, map[string][]referenceframe.Input) ([]node, error)
 	opt() *plannerOptions
 	sample(node, int) (node, error)
 }
 
-type plannerConstructor func(frame.Frame, *rand.Rand, logging.Logger, *plannerOptions) (motionPlanner, error)
+type plannerConstructor func(referenceframe.FrameSystem, *rand.Rand, logging.Logger, *plannerOptions) (motionPlanner, error)
 
 // PlanRequest is a struct to store all the data necessary to make a call to PlanMotion.
 type PlanRequest struct {
 	Logger             logging.Logger
-	Goal               *frame.PoseInFrame
-	Frame              frame.Frame
-	FrameSystem        frame.FrameSystem
+	Goal               *referenceframe.PoseInFrame
+	Frame              referenceframe.Frame
+	FrameSystem        referenceframe.FrameSystem
 	StartPose          spatialmath.Pose
-	StartConfiguration map[string][]frame.Input
-	WorldState         *frame.WorldState
+	StartConfiguration map[string][]referenceframe.Input
+	WorldState         *referenceframe.WorldState
 	BoundingRegions    []spatialmath.Geometry
 	Constraints        *Constraints
 	Options            map[string]interface{}
+	allGoals           PathStep // TODO: this should replace Goal and Frame
```

Comment:
It will indeed be eventually capitalized.

I don't think there's a need for this to take an array; when pre-planning is fully operational, calling pre-plans iteratively will yield the same result while giving the caller extra control.